config/sprayproxy.yaml

```diff
@@ -0,0 +1,70 @@
+# Copyright © 2023 The Spray Proxy Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sprayproxy
+  namespace: redhat-sprayproxy
+  labels:
+    app.kubernetes.io/name: sprayproxy
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: sprayproxy
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: sprayproxy
+    spec:
+      serviceAccountName: redhat-sprayproxy
+      containers:
+        - name: sprayproxy
+          image: ko://github.com/redhat-appstudio/sprayproxy
+          args:
+            - server
+          env:
+            - name: SPRAYPROXY_SERVER_HOST
+              valueFrom:
+                secretKeyRef:
+                  name: sprayproxy-config
+                  key: HOST
+                  optional: true
+            - name: SPRAYPROXY_SERVER_PORT
+              valueFrom:
+                secretKeyRef:
+                  name: sprayproxy-config
+                  key: PORT
+                  optional: true
+            - name: SPRAYPROXY_SERVER_BACKENDS
+              valueFrom:
+                secretKeyRef:
+                  name: sprayproxy-config
+                  key: BACKENDS
+                  optional: false
+            - name: SPRAYPROXY_SERVER_INSECURE_SKIP_TLS_VERIFY
+              valueFrom:
+                secretKeyRef:
+                  name: sprayproxy-config
+                  key: INSECURE
+                  optional: true
+          resources:
+            limits:
+              memory: "128Mi"
+              cpu: "500m"
+
+---
+apiVersion: v1
+kind: Service
```

Comment:
TODO for @gabemontero - add a second Service for the metrics (which will be referenced by the service monitor).

We may also want to check with other teams if they are using the kube-rbac-proxy sidecar to add authorization to the metrics endpoint.

Reply:
I have added the ports in the same service, is that okay or it needs to be a separate service?

Reply:
I'm okay keeping it for now. We can fix once we understand what is needed for the service monitor.